### PR TITLE
Valkyrize batch editing

### LINKED ISSUE #5797: Editing batch of works generates "NoMethodError in Hyrax::BatchEditsController#edit" error
### Descriptive summary

On nurax-pg, editing a batch of works displays a "NoMethodError in Hyrax::BatchEditsController#edit" error

### Expected behavior

Selecting multiple works on the Works page and clicking "Edit Selected" should bring me to an edit page

### Actual behavior

Selecting multiple works on the Works page and clicking "Edit Selected" generates an error message:

Screen Shot 2022-08-12 at 10 54 55 AM

### Steps to reproduce the behavior

1. On nurax-pg, navigate to Dashboard -> Works
2. Select one or more works on the Works page
3. Click the "Edit Selected" button
4. Observe error message

### Related work

#5461 Issue with initial report of batch work edit problem but error message was different

### PR DESCRIPTION
**Pull request by cjcolvar**
This PR has changes that allow batch editing to work in valkyrized hyrax.
Changes proposed:
- Add `ResourceBatchEditForm` based on `BatchEditForm` and `ResourceForm` to be used when `use_valkyrie` is true
- Handle updating each batch item using new form and work update transaction

This PR is built on top of #5863 so will need to be rebased after it is merged.
Fixes #5797

### CODE REVIEW

File: app/controllers/concerns/hyrax/works_controller_behavior.rb [hunk 1]
```diff
@@ -189,7 +189,8 @@ def create_valkyrie_work
           'work_resource.add_to_parent' => { parent_id: params[:parent_id], user: current_user },
           'work_resource.add_file_sets' => { uploaded_files: uploaded_files, file_set_params: params[hash_key_for_curation_concern][:file_set] },
           'change_set.set_user_as_depositor' => { user: current_user },
-          'work_resource.change_depositor' => { user: ::User.find_by_user_key(form.on_behalf_of) }
+          'work_resource.change_depositor' => { user: ::User.find_by_user_key(form.on_behalf_of) },
```
Comment (no-reply) [inline app/controllers/concerns/hyrax/works_controller_behavior.rb:192]: is this related here, or cruft from your other PR?

Comment (no-reply) [reply app/controllers/concerns/hyrax/works_controller_behavior.rb:192]: i think this is the cause of the test failures.

**Review by alishaevn — CHANGES_REQUESTED** [review]
the koppie app is working, but the hyrax app is not. is that your expectation? or do you have other work from this branch that hasn't been pushed?

**Comment by cjcolvar** [comment]
Sorry, it has been a long time since I've worked on this so I don't recall the exact state where I left it.  I checked and I believe that I've pushed everything for this branch.  I don't have any idea when I'll be working on this again so anyone is free to take what's useful and carry the effort forward.

**Review by alishaevn — DISMISSED** [review]

**Comment by tpendragon** [comment]
Closed, this got fixed in #6148

**Pull request closed without merging**